stumpy/mstump.py

```diff
@@ -1100,8 +1229,27 @@ def mstump(T, m, include=None, discords=False, normalize=True, p=2.0):
     T_A = T
     T_B = T_A
 
-    T_A, M_T, Σ_T, T_subseq_isconstant = core.preprocess(T_A, m)
-    T_B, μ_Q, σ_Q, Q_subseq_isconstant = core.preprocess(T_B, m)
+    T_A = core._preprocess(T_A)
+    T_B = core._preprocess(T_B)
+
+    T_A_subseq_isconstant = T_subseq_isconstant
+    if T_A_subseq_isconstant is None or callable(T_A_subseq_isconstant):
+        T_A_subseq_isconstant = [T_A_subseq_isconstant] * T_A.shape[0]
+
+    T_A_subseq_isconstant = np.array(
+        [
+            core.rolling_isconstant(T_A[i], m, T_A_subseq_isconstant[i])
+            for i in range(T_A.shape[0])
+        ]
+    )
+    T_B_subseq_isconstant = T_A_subseq_isconstant
+
+    T_A, M_T, Σ_T, T_subseq_isconstant = core.preprocess(
```

Comment:
A little bit of code smell here. Feels a tiny bit redundant to do `_preprocess` at the top and then `preprocess` (which has the `_preprocess` step). I understand why it needs to be done this way

Reply:
> A little bit of code smell here. Feels a tiny bit redundant to do `_preprocess` at the top and then `preprocess` (which has the `_preprocess` step). I understand why it needs to be done this way

One way to resolve this code smell is to put this code inside the `core.preprocess`. Note that this function, i.e. `core.preprocess`, already supports multi-dim time series. So, we can add support for multi-isconstant "T_subseq_isconstant", when provided as type `list` by the user.

What do you think @seanlaw ?

Reply:
Sure. As long as it is still readable